File: backend/src/main/java/unam/ciencias/ids/playbit/models/MatchGame.java
package unam.ciencias.ids.playbit.models;


import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.OneToOne;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
@Entity
@Table(name = "matchgame")
public class MatchGame {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id")
    private int id;

    @Enumerated(EnumType.STRING)
    @Column(name = "matchstatus")
    private MatchStatus matchStatus;


    @OneToOne(mappedBy = "opponentOneMatchGame")
    private ParticipantMatchGameResult opponentOneResult;
    // @Column(name = "opponent1id")
    // private int opponentOneResultId;
    
    @OneToOne(mappedBy = "opponentTwoMatchGame")
    private ParticipantMatchGameResult opponentTwoResult;
    // @Column(name = "opponent2id")
    // private int opponentTwoResultId;

    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "stage_id")
    private Stage stage;

    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "match_id")
    private Match match;

    @Column(name = "number")
    private int number;
}

File: frontend/src/tranformers/MatchGame.ts
import { MatchGame } from "brackets-model";
import MyMatchGame from "../types/MyMatchGame";
import MatchGameResultT from "./MatchGameResult";

class MatchGameT {

    resultT = new MatchGameResultT;

    to (value:  MyMatchGame): MatchGame {
        return {
            id: value.id,
            parent_id: value.match.id,
            stage_id: value.stage.id,
            status: value.matchStatus as number,
            opponent1: this.resultT.to(value.opponentOneResult),
            opponent2: this.resultT.to(value.opponentTwoResult),
            number: value.number
        }
    }

    from (value: MatchGame): MyMatchGame {
        return {
            id: value.id as number,
            matchStatus: value.status as number,
            opponentOneResult: this.resultT.from(value.opponent1),
            opponentTwoResult: this.resultT.from(value.opponent2),
            stage: {
                id: value.stage_id as number
            },
            number: value.number,
            match: {
                id: value.parent_id as number
            }
        }
    }
}

export default MatchGameT